### ISSUE TITLE
Color palette accessibility

### ISSUE THREAD

**Issue by domyen** [storybookjs/design-system#120]
[Reported by @Hypnosphi](https://github.com/storybookjs/storybook/issues/8935#event-2826065589) 

**Goal**
Audit and update color palette to something that's perceptually similar to the current palette but with improved accessibility.

Examples: 
- 🔴 White text on blue background 
- 🔴 Green text on green background in badge

Reference 
- [Colorbox.io](https://www.colorbox.io/) from Lyft 
- Stripe's recent [color system article](https://stripe.com/blog/accessible-color-systems) to inform this.

Would love some help with this if you have experience with perceptual colors + a11y. 🙏

**Comment by livekeen** [comment]
Accessibility is kind of a vague catch-all phrase but here's where I think it matters in regards to Storybook. Basically the EU has a directive to officially requiring its member states to abide by WCAG 2.1 Level AA standards.

It has a number of expectations that follow it beyond accessible colours but let's drill into that detail (which can be found here: https://www.w3.org/WAI/WCAG21/quickref/#contrast-minimum).

Basically to start adjusting the colours in the colour contrast guidelines you need to identify a few things:
- is this colour used for an icon, text, and/or a background?
- when this colour is used for an icon or text what background colour is it on?

I'll submit a first pass in a little bit but I just wanted to cover the basic thinking so that there's some logic to the madness I'll submit shortly.

**Comment by livekeen** [comment]
[Palette preview](https://imgur.com/oelmlfC)

### theme.color.primary - Coral
🆕 #FF4785 ➡️ #DE2865
🟢 [Text on white](https://color.review/check/DE2865-FFFFFF)
🟢 [Text on black](https://color.review/check/DE2865-000000)

### theme.color.secondary - Blue
🆕 #1EA7FD ➡️ #177CBB
🟢 [Text on white](https://color.review/check/177CBB-FFFFFF)
🟢 [Text on black](https://color.review/check/177CBB-000000)

### theme.color.tertiary - Grey
🆕 #DDDDDD ➡️ #767676
🟢 [Text on white](https://color.review/check/767676-FFFFFF)
🟢 [Text on black](https://color.review/check/767676-000000)

### theme.color.positive - Green
🆕 #66BF3C ➡️ #47852A
🟢 [Text on white](https://color.review/check/47852A-FFFFFF)
🟢 [Text on black](https://color.review/check/47852A-000000)

### theme.color.warning - Ochre
#E69D00*
🔴 [Text on white](https://color.review/check/E69D00-FFFFFF)
🟢 [Text on black](https://color.review/check/E69D00-000000)
*#9E6C00 would work against white but I'm not sure if it looks appropriate for a warning colour. It may be advisable to revisit this colour choice altogether rather than finding a perceptually similar option in this case

### theme.color.negative - Red
🆕 #FF4400 ➡️ #DB3B00
🟢 [Text on white](https://color.review/check/DB3B00-FFFFFF)
🟢 [Text on black](https://color.review/check/DB3B00-000000)

### theme.color.orange - Orange
🆕 #FC521F ➡️ #D2471D
🟢 [Text on white](https://color.review/check/D2471D-FFFFFF)
🟢 [Text on black](https://color.review/check/D2471D-000000)

### theme.color.gold - Gold
🆕 #FFAE00 ➡️ #C68700
🔴 [Text on white](https://color.review/check/C68700-FFFFFF)
🟢 [Icon or <18pt text on white](https://color.review/check/C68700-FFFFFF)
🟢 [Text on black](https://color.review/check/C68700-000000)

### theme.color.green - Green
🆕 #66BF3C ➡️ #46842A
🟢 [Text on white](https://color.review/check/46842A-FFFFFF)
🟢 [Text on black](https://color.review/check/46842A-000000)

### theme.color.seafoam - Seafoam
🆕 #37D5D3 ➡️ #228281
🟢 [Text on white](https://color.review/check/228281-FFFFFF)
🟢 [Text on black](https://color.review/check/228281-000000)

### theme.color.purple - Purple
🆕 ##6F2CAC ➡️ #945CC7**
🟢 [Text on white](https://color.review/check/6F2CAC-FFFFFF)
🟢 [Text on black](https://color.review/check/6F2CAC-000000)
**Perceptually similar in my mind means the colour visually looks like it belongs in the same family as the original. I suspect this suggested colour may be a bit off from that criteria but I think the alternative option would be to limit the usage this colour to certain scenarios.

### theme.color.ultraviolet - Ultraviolet
#2A0481**
🟢 [Text on white](https://color.review/check/2A0481-FFFFFF)
🔴 [Text on black](https://color.review/check/2A0481-000000)
**Based on my understanding of perceptually similar I feel it would be better to just identify what conditions this colour doesn't meet colour contrast.

----

Please review this first pass when you have a chance and let me know whether I understood the criteria correctly.
